test/object-store/sync/app.cpp

```diff
@@ -805,6 +806,13 @@ TEST_CASE("app: remote mongo client", "[sync][app]") {
         {"breed", "french bulldog"},
     };
 
+    auto cat_id_string = ObjectId::gen().to_string();
```

Comment:
FYI, we also have `util::random_string(N)`. It may be a slightly stronger test if the id isn't in ObjectID format (not blocking).